### PR TITLE
Feature/ERC20-token-transfer-across-chains

### PR DESCRIPTION
**Pull request by vivekjain23**
# Description of change

This Pull Request updates the documentation for compiling contracts in the LayerZero OFT V2 repository. The changes aim to provide instructions for transferring standard ERC20 contracts and custom wSMR/wIOTA tokens across chains.

Resolved issue [404](https://github.com/iotaledger/devx/issues/404)

- [X] I have followed the [contribution guidelines](https://github.com/iota-wiki/iota-wiki/blob/main/.github/CONTRIBUTING.md) for this project
- [X] I have performed a self-review of my own changes
- [X] I have made sure that added/changed links still work
- [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: docs/build/isc/v1.0.0-rc.6/docs/how-tos/send-ERC20-across-chains.md [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+:::
+
+### Procedure to Send Tokens from Source Chain to Destination Chain and Vice Versa
+
+For existing ERC20 tokens that involve both the OFTAdapter contract (on the source chain) and the OFT contract (on the destination chain), the token sending procedure is as follows:
+
+1. **Approve Tokens**: The sender approves their ERC20 tokens for the OFTAdapter contract.
+2. **Estimate Fee**: The sender calls the function `quoteSend()` of the OFTAdapter contract to estimate the cross-chain fee to be paid in native tokens on the source chain.
+3. **Send Tokens**: The sender calls the function `send()` of the OFTAdapter contract to transfer tokens from the source chain to the destination chain.
+4. **Optional - Wait for Finalization**: The sender can wait for transaction finalization on the destination chain using the library [@layerzerolabs/scan-client](https://www.npmjs.com/package/@layerzerolabs/scan-client#example-usage).
```
Comment (lucas-tortora) [inline docs/build/isc/v1.0.0-rc.6/docs/how-tos/send-ERC20-across-chains.md:38]:
```suggestion
4. **(Optional) Wait for Finalization**: The sender can wait for transaction finalization on the destination chain using the library [@layerzerolabs/scan-client](https://www.npmjs.com/package/@layerzerolabs/scan-client#example-usage).
```

File: docs/build/isc/v1.0.0-rc.6/docs/how-tos/send-ERC20-across-chains.md [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+
+To send back the OFT-wrapped tokens on the destination chain to the source chain, the procedure is similar, except that the approval step is not needed:
+
+1. **Estimate Fee**: The sender calls the function `quoteSend()` of the OFT contract to estimate the cross-chain fee to be paid in native tokens on the sender chain.
+2. **Send Tokens**: The sender calls the function `send()` of the OFT contract to transfer tokens from the destination chain back to the source chain.
+3. **Optional - Wait for Finalization**: The sender can wait for transaction finalization on the destination chain using the library `@layerzerolabs/scan-client`.
+
+
+
+#### Sample Solidity code for OFTAdater and OFT contracts in folder `contracts-standard`
```
Comment (lucas-tortora) [inline docs/build/isc/v1.0.0-rc.6/docs/how-tos/send-ERC20-across-chains.md:48]:
```suggestion
## Sample Solidity code for OFT Adapter and OFT contracts in the folder `contracts-standard`
```

File: docs/build/isc/v1.0.0-rc.6/docs/how-tos/send-ERC20-across-chains.md [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+- Scripts for:
+  - Deploy OFTAdapter and OFT contracts
+  - Set trusted peer
+  - Set enforced options
+  - Send tokens from source chain to destination chain and vice versa
```
Comment (lucas-tortora) [inline docs/build/isc/v1.0.0-rc.6/docs/how-tos/send-ERC20-across-chains.md:53]:
```suggestion
The [contracts-standard](https://github.com/iota-community/layerzero-oft-v2-utils/tree/main/contracts-standard) contains scripts to:

- Deploy the OFT Adapter and OFT contracts.
- Set your trusted peers.
- Set enforced options.
- Send tokens from the source chain to the destination chain, and vice versa.
```

File: docs/build/isc/v1.0.0-rc.6/docs/how-tos/send-ERC20-across-chains.md [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+
+## Installation
+```
+$ yarn
+```
+
+## Compile contracts
```
Comment (lucas-tortora) [inline docs/build/isc/v1.0.0-rc.6/docs/how-tos/send-ERC20-across-chains.md:60]:
```suggestion
## Compile the Contracts
```

File: docs/build/isc/v1.0.0-rc.6/docs/how-tos/send-ERC20-across-chains.md [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+
+- In the standard implementation for ERC20, copy the folder `contracts-standard`to `contracts` and `contracts-wiota` to `contracts` for custom implementation. Then, run the command:
```
Comment (lucas-tortora) [inline docs/build/isc/v1.0.0-rc.6/docs/how-tos/send-ERC20-across-chains.md:62]:
```suggestion
If you want to use the standard implementation for ERC20, copy the [`contracts-standard`](https://github.com/iota-community/layerzero-oft-v2-utils/tree/main/contracts-standard) folder to `contracts`. If you want to use a custom implementation, copy the  [`contracts-wiota`](https://github.com/iota-community/layerzero-oft-v2-utils/tree/main/contracts-wiota) to `contracts`. Then, run the following command to compile the contracts:
```

File: docs/build/isc/v1.0.0-rc.6/docs/how-tos/send-ERC20-across-chains.md [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+```
+  $ yarn compile
+``` 
+
+## Configuration
+
+The config is specified in the template file `.env.example` that needs to be copied to another file `.env`.
+
+
+## Deploy contracts
```
Comment (lucas-tortora) [inline docs/build/isc/v1.0.0-rc.6/docs/how-tos/send-ERC20-across-chains.md:73]:
```suggestion
### Deploy the Contracts
```

File: docs/build/isc/v1.0.0-rc.6/docs/how-tos/send-ERC20-across-chains.md [hunk 7]
```diff
@@ -0,0 +1,174 @@
...
+
+```
+$ npx hardhat run scripts/deploy_oft.ts --network bnbTestnet
+Deployed MyOFT contract address: 0xCc337C2e69F4Eb8EaBcf632a1fC5B8F729dC47F1
+Done in 6.68s.
+```
+
+## Set trusted peer (Optinal)
+
+### On OFTAdapter (source chain, e.g. Sepolia)
```
Comment (lucas-tortora) [inline docs/build/isc/v1.0.0-rc.6/docs/how-tos/send-ERC20-across-chains.md:110]:
```suggestion
### On the OFT Adapter
```

**Comment by vivekjain23** [comment]
Thanks for the valuable insights, I have added the changes in the new commit.

File: docs/build/isc/v1.0.0-rc.6/docs/how-tos/send-ERC20-across-chains.md [hunk 8]
```diff
@@ -0,0 +1,174 @@
...
+3. **Send Tokens**: The sender calls the function `send()` of the OFTAdapter contract to transfer tokens from the source chain to the destination chain.
+4. **Optional - Wait for Finalization**: The sender can wait for transaction finalization on the destination chain using the library [@layerzerolabs/scan-client](https://www.npmjs.com/package/@layerzerolabs/scan-client#example-usage).
+
+To send back the OFT-wrapped tokens on the destination chain to the source chain, the procedure is similar, except that the approval step is not needed:
+
+1. **Estimate Fee**: The sender calls the function `quoteSend()` of the OFT contract to estimate the cross-chain fee to be paid in native tokens on the sender chain.
+2. **Send Tokens**: The sender calls the function `send()` of the OFT contract to transfer tokens from the destination chain back to the source chain.
+3. **Optional - Wait for Finalization**: The sender can wait for transaction finalization on the destination chain using the library `@layerzerolabs/scan-client`.
```
[the remark this answers is not in the document]
Comment (vivekjain23) [reply docs/build/isc/v1.0.0-rc.6/docs/how-tos/send-ERC20-across-chains.md]: Sure Lucas, I have added code samples for all the above points required.

**Comment by Dr-Electron** [comment]
Can you do me one more favour? 😅 
Please rebase on top of `isc-update` there you will also have to duplicate your content as we started to version wasp.
So there is a 1.1 and 1.3 folder.

**Review by lucas-tortora — APPROVED** [review]

**Review by lucas-tortora — APPROVED** [review]

**Pull request merged**